macformula/hil
# Adds Mcap generation and refactors tracer

Credit to @ksmart1403 and @floralandfaded for doing all this!

Closes #44, #45, #46, #47, #47, #49

Refactors tracer to add Mcap generation that can plot CAN messages on https://app.foxglove.dev/

The following are foxglove templates to view the 3 different Messages sent in cmd/tracetest/main.go.
[Pack_SOC.json](https://github.com/macformula/hil/files/14630611/Pack_SOC.json)
[Contactor_Feedback.json](https://github.com/macformula/hil/files/14630612/Contactor_Feedback.json)
[Pack_State.json](https://github.com/macformula/hil/files/14630613/Pack_State.json)

### How to test

- `go build cmd/tracetest/main.go`
- `sudo ./main`

Then to retrieve the logs
- `cd /opt/macfe/traces`

and there you should see a .asc and a .mcap

You can download the .mcap and visualize it on the templates above
